### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by swartzlib7** [nextcloud/desktop#6099]
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

# Data Loss during External Storage Disconnect

Firstly, thank you for the awesome Desktop Sync client and Nextcloud in general. I love it. We do however have a small problem to be addressed rather rapidly.

This issue is fairly serious and hits at confidence in sync client behavior which will cause loss of data locally due to a common technical issue of having your external storage device fail or be turned off without turning while still running your Nextcloud instance.

> When the Nextcloud external storage device becomes unavailable, the mount points configured on Nextcloud are unavailable in the back-end with a visible error, but the sync client remains operational - data deletion of all data from the configured local storage will result!

## Hardware and Configuration
* NextcloudPi on a Raspberry Pi 4.
* 2TB raid via USB 3 for external storage.
* Nextcloud v 27.0.1.
* Linux sync client v3.10.0 (KDE).

## Situation
* My raid storage device didn't start due to a suspected power loss during the last 10 days.
* I was running Nextcloud v26.x.x (latest) at this time and the instance stopped functioning completely after power was restored.
* Usually, I manually run the desktop sync client as needed - I didn't use the sync client until I got the instance back online.
* I upgraded to Nextcloud v27.0.1 which brought my instance back online.
* I was working locally on my synced folder for about a week with the sync client **inactive**.

## Result
* I opened the sync client after Nextcloud v27.0.1 was back online, but the external storage was still offline.
* Data loss resulted when the sync client started. All files and folders from the local storage location was removed.
* Inspection of the Nextcloud External Storage configuration showed warnings present about mounting failures of the external storage device, but these warnings were not taken into account by the sync client.
* After mounting the storage device the sync client is now again syncing data locally.
* Any new files created during the offline period and present in the configured sync folder for external storage has been deleted by the sync client and lost.

## Suggestion
* Inspect external storage mounting status before deleting files.
* When deleting files send them to the recycle bin or something similar without simply unlinking files from disc.
* Add a data threshold for deletions e.g. anything over 100 MB of data will require confirmation.

### Steps to reproduce

1. Run Nextcloud v27.0.1.
2. Have external storage connected.
3. Run Sync client with external storage folder to local machine configured and let it sync.
4. Stop/Pause sync client.
5. Create new files in local external storage folder (not on storage device).
6. Disconnect external storage from Nextcloud device (unmount and turn it off while leaving Nextcloud server running).
7. Resume sync client and watch all existing and new data disappear without warning.

### Expected behavior

* Sync client to prompt for large delete operation.
* Sync client to check external storage mounting status before deleting.
* Move files to a "deleted" folder or something such as recycle bin before just removing them.

### Which files are affected by this bug

no_idea.log

### Operating system

Linux

### Which version of the operating system you are running.

Pop!_OS 22.04 LTS

### Package

Appimage

### Nextcloud Server version

27.0.1

### Nextcloud Desktop Client version

v3.10.0 (KDE)

### Is this bug present after an update or on a fresh install?

Fresh desktop client install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [X] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by mgallien** [comment]
@swartzlib7 
I cannot reproduce this with a setup using NFS external storage and versions of server and desktop files client matching your versions.
I will need a debug archive from your client to debug further

**Comment by swartzlib7** [comment]
Hi Matthieu,

thank you for getting back to my issue here.

I updated my client AppImage since the last issue, but the logs might still be available if this is stored in my home directory.

Can you tell me where I could find it? I'm running linux.

**Comment by mgallien** [comment]
@swartzlib7 the logs are by default (unless you changed the settings) kept for 24 hours
I would assume you no longer have them.
what kind of external storage are you using ?

**Comment by mgallien** [comment]
so the logs are by default located in
`~/.config/nextcloud/logs`

**Comment by swartzlib7** [comment]
Yeah those logs are long gone. Thank you for helping me with this!
So the drive I use is connected via USB3 and configured as an external storage device.
The external device is a Yottamaster 2 Bay RAID configured for Raid 1.

**Comment by mgallien** [comment]
@swartzlib7 I am investigating
the nominal behavior is to skip folders that are from external storage but unmounted at time of sync
I tested with NFS external storage without having been able to reproduce the issue
I was running the same versions you were running
I will try again with other types of external storage just in case

**Comment by l0drex** [comment]
If I understood correctly, I just ran into the same issue. The desktop client deleted a bunch of files with the message `File was deleted (File with name //Festplatte could not be located)`. `//Festplatte` would be the mount point of my external storage in Nextcloud.

[Here](https://hoffmanns.cloud/s/KdE65HmsApoK6S9) is the debug archive exported from the desktop client (was too big for GitHub). Nextcloud desktop is installed through pacman.

<details>
<summary>My system information</summary>

Operating System: Arch Linux 
KDE Plasma Version: 5.27.9
KDE Frameworks Version: 5.111.0
Qt Version: 5.15.11
Kernel Version: 6.6.1-zen1-1-zen (64-bit)
Graphics Platform: X11
Processors: 12 × Intel® Core™ i5-10400F CPU @ 2.90GHz
Memory: 15.5 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 2080/PCIe/SSE2
Product Name: H570 Phantom Gaming 4
</details>

**Comment by l0drex** [comment]
This is a rather urgent issue, and I would love to see this addressed quickly. If you need anything else, please let me know.

**Comment by swartzlib7** [comment]
> @swartzlib7 I am investigating the nominal behavior is to skip folders that are from external storage but unmounted at time of sync I tested with NFS external storage without having been able to reproduce the issue I was running the same versions you were running I will try again with other types of external storage just in case

Thank you for this and to everyone else who also confirmed the same thing happening on their clients.

I have since moved away from the NextCloud Pi solution and the Pi4 I used. I am now on different hardward and the NextCloud AIO solution. So far this is far better and also a bit faster due to the new hardware.

That said, I still have this in the back of my mind and would love to see a fix with clear UI / log indication of an event where an external mount goes missing and it's files being "orphaned" or left alone in-place rather than kicked out into a "black hole".

Sincerely,
Stephen

**Comment by theroch** [comment]
My nextcloud desktop client (3.12.0 windows) also deleted all my local files after the SMB share becomes unavailable for nextcloud (27.1.7) because of a software package update for samba was installed.

I think it is almost not a good behavior of a file client to delete GBs of data, without asking the user.
Please implement a dialog to ask the user for such situations.

**Comment by bigretromike** [comment]
> My nextcloud desktop client (3.12.0 windows) also deleted all my local files after the SMB share becomes unavailable for nextcloud (27.1.7) because of a software package update for samba was installed.
> 
> I think it is almost not a good behavior of a file client to delete GBs of data, without asking the user. Please implement a dialog to ask the user for such situations.
was your SMB mount point added as external storage ? 
Because there was a fix for something like this some time ago: https://github.com/nextcloud/server/pull/39707  maybe its recursing issue ?

**Comment by joshtrichards** [comment]
Related: #1984
